perllib/FixMyStreet/App/Controller/Report.pm

```diff
@@ -660,6 +662,59 @@ sub map :Chained('id') :Args(0) {
     $c->res->body($image->{data});
 }
 
+sub confirmation : Path('confirmation') : Args(1) {
+    my ( $self, $c, $id ) = @_;
+
+    # First of all check that the report ID is valid
+    my $report = FixMyStreet::DB->resultset('Problem')->find( { id => $id } );
+    unless ( $report ) {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+
+    # Now verify the token we've been given is correct
+    my $token_param = $c->get_param('token') || "";
+    my $token = $c->forward('/report/get_confirmation_token', [ $report ]);
+    unless ( $token eq $token_param ) {
+        $c->detach( '/page_error_404_not_found', [] );
+    }
+
+    # If the token is valid but expired then may as well be helpful and bounce
+    # the user to report page rather than 404.
+    # (NB the report may still be unconfirmed, but end result is the same - a 404)
+    my $cutoff = DateTime->now()->subtract( minutes => 1 )->epoch;
+    my $timestamp = $report->confirmed ? $report->confirmed : $report->created;
+    if ( $timestamp->epoch < $cutoff ) {
+        return $c->res->redirect($report->url);
+    }
+
+    # We're now confident the user is allowed to view this report so stick it on
+    # the stash and load up the correct template.
+    $c->stash->{problem} = $report;
```

Comment:
What's this line for? Looks like New.pm only ever uses `{report}`. Is it the header navigation template - so has this not been working before? Just curious.

Reply:
Yep, it's for `navigation/_report.html` and e.g. BANES and the `post_report_report_problem_link` feature.